src/backend/linux_raw/net/syscalls.rs

```diff
@@ -178,36 +171,28 @@ pub(crate) fn accept_with(fd: BorrowedFd<'_>, flags: SocketFlags) -> io::Result<
 pub(crate) fn acceptfrom(fd: BorrowedFd<'_>) -> io::Result<(OwnedFd, Option<SocketAddrAny>)> {
     #[cfg(not(any(target_arch = "x86", target_arch = "s390x")))]
     unsafe {
-        let mut addrlen = core::mem::size_of::<sockaddr_storage>() as socklen_t;
-        let mut storage = MaybeUninit::<sockaddr_storage>::uninit();
+        let mut addr = SocketAddrBuf::new();
         let fd = ret_owned_fd(syscall!(
             __NR_accept,
             fd,
-            &mut storage,
-            by_mut(&mut addrlen)
+            &mut addr.storage,
+            by_mut(&mut addr.len)
         ))?;
-        Ok((
-            fd,
-            maybe_read_sockaddr_os(&storage.assume_init(), addrlen.try_into().unwrap()),
```

Comment:
Also, I noticed that prior to this PR, these `assume_init` calls were all undefined behavior, because the `sockaddr_storage` is not necessarily fully initialized. That's why `SocketAddrAny` contains `MaybeUninit`.